geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/util/GeoHashUtils.scala

```diff
@@ -0,0 +1,165 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.spark.jts.util
+
+import com.vividsolutions.jts.geom.{Envelope, Geometry, GeometryFactory, PrecisionModel}
+import scala.collection.BitSet
+import scala.collection.immutable.{BitSet => IBitSet}
+
+object GeoHashUtils {
+
+  def encode(geom: Geometry, precision: Int): String = {
+    GeoHash.encode(geom.getCentroid.getX, geom.getCentroid.getY, precision)
```

Comment:
centroid can sometimes return NaN - see https://github.com/locationtech/geomesa/blob/geomesa_2.11-2.0.0-m.1/geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Conversions.scala#L45
Also, probably better to only call `.getCentroid` once, as i'm not sure if the result is cached or re-calculated. (if you can determine it's cached, add a comment to that effect)